catalog/spatial/csw/spatial-csw-endpoint/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/TestCswEndpoint.java

```diff
@@ -1182,12 +1183,44 @@ public void testRetrieveProductGetRecordById()
         getRecordByIdRequest.setOutputSchema(OCTET_STREAM_OUTPUT_SCHEMA);
         setUpMocksForProductRetrieval(true);
 
-        CswRecordCollection cswRecordCollection = csw.getRecordById(getRecordByIdRequest);
+        String rangeValue = "bytes=100-";
+
+        CswRecordCollection cswRecordCollection = csw.getRecordById(getRecordByIdRequest,
+                rangeValue);
 
         assertThat(cswRecordCollection.getResource(), is(notNullValue()));
     }
 
     @Test
+    public void testRetrieveProductGetRecordByIdWithRange()
+            throws IOException, ResourceNotFoundException, ResourceNotSupportedException,
+            CswException {
+        final GetRecordByIdRequest getRecordByIdRequest = new GetRecordByIdRequest();
+        getRecordByIdRequest.setId("123");
+        getRecordByIdRequest.setOutputFormat(MediaType.APPLICATION_OCTET_STREAM);
+        getRecordByIdRequest.setOutputSchema(OCTET_STREAM_OUTPUT_SCHEMA);
+        setUpMocksForProductRetrieval(true);
+
+        CswRecordCollection cswRecordCollection = csw.getRecordById(getRecordByIdRequest,
+                "bytes=100-");
+
+        assertThat(cswRecordCollection.getResource(), is(notNullValue()));
```

Comment:
Since the resource is mocked there really isn't a way to validate that the range was interpreted/respected. However, integration tests can definitely verify that. I did add unit tests to specifically test the range method.